IAV-1825 Health Questionnaire (slider update)

juhog opened the pull request:
Summary:
- jQuery UI better css file (with only structure, no theme)
- Base style 'visually-hidden'
- Slider update (js, scss, twig)

I suggest taking a look at the commits one by one to see what this PR contains.

I'm not sure if jquery-ui-touch-punch library should be added to the styleguide to fix mobile sliders. Currently it's not part of this PR.

----

Screenshot from local. Github public version of the slider won't work until slider.js related problem is solved (mentioned in Slack).

<img width="813" alt="screen shot 2016-12-14 at 23 27 09" src="https://cloud.githubusercontent.com/assets/1115979/21190559/c9fb694a-c254-11e6-9c99-375e0ca8ba5e.png">

lauriii commented:
I think the library to add this functionality to mobile should be added as part of this PR.

juhog commented:
> I think the library to add this functionality to mobile should be added as part of this PR.

@lauriii This is now fixed. I added jQuery Touch Punch to the styleguide in this commit: https://github.com/LaakarikeskusAava/styleguide/pull/12/commits/81e268da5d6b091f2a167d1a213a5030a66ba282

There is still one problem in the slider though. The slider works in local but it doesn't work in the public github page. See this slack message: https://druid.slack.com/archives/aava-web/p1481729948000210

juhog commented:
> There is still one problem in the slider though. The slider works in local but it doesn't work in the public github page. See this slack message: https://druid.slack.com/archives/aava-web/p1481729948000210

This problem should be fixed is now. Component .js files are copied to public directory. See this commit https://github.com/LaakarikeskusAava/styleguide/pull/12/commits/fd8196b796d2c8ff7d1bde13b8bddd56a2f96ca9 and the screens below.

<img width="626" alt="screen shot 2017-03-25 at 09 56 11" src="https://cloud.githubusercontent.com/assets/1115979/24318971/2d7abbc4-1142-11e7-997b-d78a4331e5ba.png">
<img width="250" alt="screen shot 2017-03-25 at 09 57 07" src="https://cloud.githubusercontent.com/assets/1115979/24318972/2d7eb0d0-1142-11e7-8c42-1b284e76d78b.png">

---

@lauriii Is this PR ok to merge now?

juhog commented:
@lauriii Can I get a review for this PR? (There's one fix that's gonna help the new form style PR too.)

lauriii commented on styles/components/forms/slider/slider.json:
```diff
@@ -1,9 +1,5 @@
-  ]
+  "min": 0,
+  "max": 10,
+  "input": "<input type=\"range\" min=\"0\" max=\"10\" value=\"5\">"
```
Why do we have html in the json?

juhog replied:
Front-end details in Finnish.

En nyt tarkkaan muista ilman että kaivelis vanhoja tietoja.

Muistaakseni kuitenkin idea on se, että slider-komponentille annetaan form-elementti, ja slider-komponentti muuttaa sen sitten tollaseks jquery ui slideriksi. Kun slideria käyttää niin sen form-elementin arvo muuttuu. Form-elementti voi olla `type="range"` tai `type="text"` tai vaikka `<select>`.

juhog replied:
@lauriii Does the comment above seem valid? I think this is our only component that has a .json file though.

juhog replied:
@lauriii Ping. See comments above.

lauriii replied:
This seems fine, even though I still think its weird

juhog commented:
(Test comment 1 - Does this show up in Slack?)

juhog commented:
(Test comment 2 - Does this show up in Slack?)
